src/common/snippets/src/pass/collapse_subgraph.cpp

```diff
@@ -181,6 +181,8 @@ auto has_supported_in_out(const std::shared_ptr<const Node> &n) -> bool {
     auto supported = [&n](descriptor::Tensor& t) -> bool {
         // Todo: int32 isn't supported in general because i32 emitters are required for bit-exact i32 calculations in some cases
         //  So i32 is supported exclusively for transposes and broadcast
+        if (t.get_partial_shape().rank().is_dynamic())
+            return false;
```

Comment:
May I ask you to leave the comment about dynamic rank and move comment about data type to the corresponding condition please?

```suggestion
        // TODO[122585]: Need to add dynamic rank support
        if (t.get_partial_shape().rank().is_dynamic())
            return false;
        // TODO[105804]: int32 isn't supported in general because i32 emitters are required for bit-exact i32 calculations in some cases
        //  So i32 is supported exclusively for transposes and broadcast
```

My appreciate! ❤️